File: Coding Ninjas/Easy/Reverse First K elements of Queue.py
def f( q, k ):

    if k < 1:
        return

    p = q.get()
    f( q, k - 1 )
    q.put( p )



def reverseElements( q, k ):
    
    f( q, k )

    for i in range( q.size() - k ):
        q.put( q.get() )

    return q

File: Coding Ninjas/Easy/Reverse First K elements of Queue.java
import java.util.*;

public class Solution 
{
    
    public static Queue< Integer > reverseElements
    ( Queue< Integer > q, int k ) 
    {
        f( q, k );
        for ( int i = k; i < q.size(); i++ )
        {
            q.offer( q.poll() );
        }
        
        return q;
    }



    static void f( Queue< Integer > q, int k )
    {
        if ( k < 1 )
        {
            return;
        }

        int t = q.poll();
        f( q, k - 1 );
        q.offer( t );
    }

}
